packages/remix-oxygen/src/server.ts

```diff
@@ -44,6 +45,13 @@ export function createRequestHandler<Context = unknown>({
       response.headers.append('powered-by', 'Shopify, Hydrogen');
     }
 
+    if ((context?.session as HydrogenSession)?.dirty) {
```

Comment:
Would there be situations where the user would explicitly want to opt out of this? I guess we can wait, and address it if it comes up.